modules/perception/camera/tools/offline/visualizer.cc

```diff
@@ -132,6 +132,60 @@ void Visualizer::SetDirectory(const std::string &path) {
   path_ = path;
 }
 
+std::string Visualizer::type_to_string(
+  const apollo::perception::base::ObjectType type) {
+  switch (type) {
+    case apollo::perception::base::ObjectType::UNKNOWN:
+     return "UNKN";
+    case apollo::perception::base::ObjectType::UNKNOWN_MOVABLE:
+     return "U_MO";
+    case apollo::perception::base::ObjectType::UNKNOWN_UNMOVABLE:
+     return "UNMO";
+    case apollo::perception::base::ObjectType::PEDESTRIAN:
+     return "PED";
+    case apollo::perception::base::ObjectType::BICYCLE:
+     return "CYC";
+    case apollo::perception::base::ObjectType::VEHICLE:
+     return "VEH";
+    default:
+     return "WRNG";
+  }
+  return "WRNG";
+}
+
+std::string Visualizer::sub_type_to_string(
+  const apollo::perception::base::ObjectSubType type) {
```

Comment:
protobuf provides functions to convert enum to string directly. i.e. try PerceptionObstacle_SubType_Name(type)

Reply:
@ycool, Thank you for the suggestion and good to know it! However, in this PR, I reduced the string of the subtypes to have only 4 letters (e.g. TRAFFICCONE -> CONE, UNKNOWN_UNMOVABLE->UNMO, etc). If PerceptionObstacle_SubType_Name is used, the string is too long to read on the small image.